### PR TITLE
Add watched checkmark to Movies, Collections, TV Shows, Season list. Remove API generated checkmark

### PR DESCRIPTION
**Pull request by photonconvergence**
Adds watched checkmark to most library views.
Aligns unwatched episode count rectangle in Season list view.
Removes the last API generated watched checkmark (Season list view)

Fixes:
https://github.com/jellyfin/jellyfin-roku/issues/312
https://github.com/jellyfin/jellyfin-roku/issues/402
https://github.com/jellyfin/jellyfin-roku/issues/1027

Inspired by:
https://github.com/jellyfin/jellyfin-roku/pull/922
https://github.com/jellyfin/jellyfin-roku/pull/1223

### CODE REVIEW

File: components/ItemGrid/GridItem.bs [hunk 1]
```diff
@@ -44,6 +48,10 @@ sub itemContentChanged()
     if itemData = invalid then return
 
     if itemData.type = "Movie"
+        if isValid(itemData.json.UserData.Played) and itemData.json.UserData.Played = true
```
Comment (1hitsong) [inline components/ItemGrid/GridItem.bs]: Let's match the valid checks we do further down the view and confirm properties are valid all the way to what we need.

```suggestion
        if isValid(itemData.json) and isValid(itemData.json.UserData) and isValid(itemData.json.UserData.Played) and itemData.json.UserData.Played
```

File: components/ItemGrid/GridItemSmall.bs [hunk 2]
```diff
@@ -37,6 +38,10 @@ sub itemContentChanged()
 
     if not isValid(itemData) then return
 
+    if isValid(itemData.json.UserData.Played) and itemData.json.UserData.Played = true
```
Comment (1hitsong) [inline components/ItemGrid/GridItemSmall.bs]:
```suggestion
    if isValid(itemData.json) and isValid(itemData.json.UserData) and isValid(itemData.json.UserData.Played) and itemData.json.UserData.Played
```

File: components/ListPoster.bs [hunk 3]
```diff
@@ -65,6 +69,10 @@ sub itemContentChanged() as void
         end if
     end if
 
+    if isValid(itemData.json.UserData.Played) and itemData.json.UserData.Played = true
```
Comment (1hitsong) [inline components/ListPoster.bs]:
```suggestion
    if isValid(itemData.json) and isValid(itemData.json.UserData) and isValid(itemData.json.UserData.Played) and itemData.json.UserData.Played
```

**Review by 1hitsong — CHANGES_REQUESTED** [review]

**Review by 1hitsong — APPROVED** [review]
These changes look good to me.

**Review by cewert — APPROVED** [review]

**Comment by chibiconsulting** [comment]
There should be a user configuration setting to control the color & visibility of the new feature.

Having the ability to select the color of the unwatched counter and the color and/or visibility of the watched checkbox would be a great user experience improvement. With the current implementation, the library views are just a sea of blue boxes.

New settings should be in: Settings / User Interface / Libraries / General

Although could also be in: Settings / User Interface / Libraries / TV Shows as there is already a setting there to "Disable Unwatched Episodes" there; however, this should be a more global solution.

In either case, the settings would be .. / Unwatched Overlay and .. / Watched Overlay with the options being **various colors** or **no display**.

**Pull request merged**